source/include/services/cs_CrownstoneService.h

```diff
@@ -98,6 +98,9 @@ class CrownstoneService: public Service, EventListener {
 
 
 private:
+	uint8_t _keySessionDataBuffer[sizeof(session_data_t)];
```

Comment:
why not just make this of type `session_data_t` and cast it to a byte buffere where necessary?

Reply:
Right now, it's never treated as session_data_t. However, in the handleEvent EVT_SESSION_DATA_SET, we don't check the type size. So maybe it's a good idea to change that.